src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -166,9 +166,11 @@ public function __construct(array $options, \Redis|Relay|\RedisCluster $redis =
      */
     private static function initializeRedis(\Redis|Relay $redis, string $host, int $port, string|array|null $auth, array $params): \Redis|Relay
     {
-        $connect = isset($params['persistent_id']) ? 'pconnect' : 'connect';
-        $redis->{$connect}($host, $port, $params['timeout'], $params['persistent_id'], $params['retry_interval'], $params['read_timeout'], ...(\defined('Redis::SCAN_PREFIX') || \extension_loaded('relay')) ? [['stream' => $params['ssl'] ?? null]] : []);
-
+        if (!$redis->isConnected()) {
```

Comment:
What about auth, serializer option and dbindex? should we consider them already initialized?

```suggestion
        if ($redis->isConnected()) {
            return $redis;
        }
```